core/src/test/java/io/questdb/test/cutlass/line/tcp/LineTcpBootstrapTest.java

```diff
@@ -66,4 +73,69 @@ public void testDisconnectOnErrorWithWAL() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            try (final ServerMain serverMain = new ServerMain(getServerMainArgs())) {
+                serverMain.start();
+                CairoEngine engine = serverMain.getEngine();
+                try (SqlCompiler sqlCompiler = engine.getSqlCompiler();
+                     SqlExecutionContext sqlExecutionContext = TestUtils.createSqlExecutionCtx(engine)) {
+                    sqlCompiler.compile("CREATE TABLE 'betfairRunners' (\n" +
+                            "  id INT,\n" +
+                            "  runner VARCHAR,\n" +
+                            "  age BYTE,\n" +
+                            "  remarks SYMBOL CAPACITY 2048 CACHE INDEX CAPACITY 256,\n" +
+                            "  timestamp TIMESTAMP\n" +
+                            ") timestamp (timestamp) PARTITION BY MONTH WAL\n" +
+                            "DEDUP UPSERT KEYS(timestamp, id);", sqlExecutionContext);
+
+                    try (Sender sender = Sender.fromConfig("http::addr=localhost:" + serverMain.getHttpServerPort() + ";")) {
+                        for (int i = 0; i < 1; i++) {
+                            sender.table("betfairRunners").symbol("remarks", "SAw,CkdRnUp&2").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-30T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.table("betfairRunners").symbol("remarks", "Fcd-Ck1").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-24T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.table("betfairRunners").symbol("remarks", "(R8) LdRnIn військові").
+                                    stringColumn("runner", "BallyMac Fifra")
+                                    .longColumn("id", 548738)
+                                    .longColumn("age", 58)
+                                    .at(TimestampFormatUtils.parseTimestamp("2024-06-17T00:00:00Z"), ChronoUnit.MICROS);
+                            sender.flush();
+                        }
+                    }
+
+                    drainWalQueue(engine);
+
+                    CompiledQuery result = sqlCompiler.compile("select distinct runner from betfairRunners", sqlExecutionContext);
+
+                    try (
+                            RecordCursorFactory factory = result.getRecordCursorFactory();
+                            RecordCursor cursor = factory.getCursor(sqlExecutionContext)
+                    ) {
+                        cursor.hasNext();
```

Comment:
nit:
```suggestion
                        Assert.assertTrue(cursor.hasNext());
```

Reply:
this was actually an issue with test, fixed in https://github.com/questdb/questdb/pull/4876/commits/d2f1d91c1479ac80e3daf436b8124b067899e3df